src/SpeedyTransforms/spectral_transform.jl

```diff
@@ -272,48 +273,58 @@ function SpectralTransform(
     )
 end
 
+"""$(TYPEDSIGNATURES)
+Generator function for a `SpectralTransform` struct based on the spectral truncation
+given by integers `lmax` and `mmax.""" 
+SpectralTransform(
+    ::Type{NF},                                     # Number format NF
+    lmax::Integer,                                  # spectral trunction degree
+    mmax::Integer,                                  # spectral truncation order
+    vargs...; kwargs...                             # grid resolution, latitude rings on one hemisphere incl equator
+) where NF = SpectralTransform(NF, Spectrum(lmax, mmax), vargs...; kwargs...)
+    
 """$(TYPEDSIGNATURES)
 Generator function for a `SpectralTransform` struct based on the size of the spectral
 coefficients `specs`. Use keyword arguments `nlat_half`, `Grid` or `deliasing` (if `nlat_half`
 not provided) to define the grid."""
 function SpectralTransform(
-    specs::LowerTriangularArray{NF, N, ArrayType};  # spectral coefficients
+    specs::LowerTriangularArray{NF, N, ArrayType, S};  # spectral coefficients
```

Comment:
btw I ended up dropping many of these so just `field::AbstractField` and then use `eltype(field)`, `ndims(field)`, etc. instead